k8s/k8s.go

```diff
@@ -171,13 +171,22 @@ func CmdAddK8s(args *skel.CmdArgs, conf utils.NetConf, hostname string, calicoCl
 			}
 		}
 
-		// Run the IPAM plugin
-		logger.Debugf("Calling IPAM plugin %s", conf.IPAM.Type)
-		result, err = ipam.ExecAdd(conf.IPAM.Type, args.StdinData)
-		if err != nil {
-			return nil, err
+		ipAddrsNoIpam := annot["cni.projectcalico.org/ipAddrsNoIpam"]
+
+		// Call IPAM plugin if ipAddrsNoIpam annotation is not present.
+		if ipAddrsNoIpam == "" {
+			// Run the IPAM plugin
+			logger.Debugf("Calling IPAM plugin %s", conf.IPAM.Type)
+			result, err = ipam.ExecAdd(conf.IPAM.Type, args.StdinData)
+			if err != nil {
+				return nil, err
+			}
+			logger.Debugf("IPAM plugin returned: %+v", result)
+		} else {
```

Comment:
Are we worried about the fact that we do IPAM related stuff up above even when this annotation is specified?

e.g https://github.com/projectcalico/cni-plugin/pull/264/files#diff-b396ca928d5bd0e161acd283b56ead9fR84

and https://github.com/projectcalico/cni-plugin/pull/264/files#diff-b396ca928d5bd0e161acd283b56ead9fR124?

Probably not because it's a kinda weird case, but it does mean we'll be doing more work than we need to.